commands/user.go

```diff
@@ -131,283 +164,445 @@ func userGroup(b *bot.Bot) {
 	})
 }
 
+// groups shows the full list of crosspost groups.
 func groups(b *bot.Bot) func(ctx *gumi.Ctx) error {
 	return func(ctx *gumi.Ctx) error {
+		// Assigns current discord user.
+		// Returns error if no user is found.
 		user, err := b.Store.User(context.Background(), ctx.Event.Author.ID)
 		if err != nil {
 			return err
 		}
 
+		// Prepares new embed builder to list crosspost groups.
 		locale := messages.UserGroupsEmbed(ctx.Event.Author.Username)
 		eb := embeds.NewBuilder()
 
+		// Builds embed title and description.
 		eb.Title(locale.Title)
 		eb.Description(locale.Description)
 
+		// Iterates and constructs crosspost groups into groupList.
+		var groupList [2][][2]string
```

Comment:
Could you please explain to me what this array does, it's a bit confusing

Reply:
First array is to seperate groups into Pairs and normal Groups.
Second, as a slice, is to get the full lists of both Pairs and Groups.
Third array is to store the embed name and description of each crosspost group.